frame/ranked-collective/src/lib.rs

```diff
@@ -683,5 +662,49 @@ pub mod pallet {
 			}
 			Ok(())
 		}
+
+		/// Decrement the rank of an existing member by one. If the member is already at rank zero,
+		/// then it is removed entirely.
+		///
+		/// A `maybe_max_rank` may be provided to check that the member does not get demoted beyond
+		/// a certain rank. Is `None` is provided, then the rank will be decremented without checks.
+		pub fn do_demote_member(who: T::AccountId, maybe_max_rank: Option<Rank>) -> DispatchResult {
+			let mut record = Self::ensure_member(&who)?;
+			let rank = record.rank;
+			if let Some(max_rank) = maybe_max_rank {
+				ensure!(max_rank >= rank, Error::<T, I>::NoPermission);
+			}
+			Self::remove_from_rank(&who, rank)?;
+			let maybe_rank = rank.checked_sub(1);
+			match maybe_rank {
+				None => {
+					Members::<T, I>::remove(&who);
+					Self::deposit_event(Event::MemberRemoved { who, rank: 0 });
+				},
+				Some(rank) => {
+					record.rank = rank;
+					Members::<T, I>::insert(&who, &record);
+					Self::deposit_event(Event::RankChanged { who, rank });
+				},
+			}
+			Ok(())
+		}
+
+		/// Remove the member entirely.
+		///
+		/// A `maybe_max_rank` may be provided to check that the member does not get removed from
+		/// a certain rank. if `None` is provided, then the rank will be decremented without checks.
+		pub fn do_remove_member(who: T::AccountId, maybe_max_rank: Option<Rank>) -> DispatchResult {
+			let MemberRecord { rank, .. } = Self::ensure_member(&who)?;
```

Comment:
`remove_member` already calls this. We should come up with a way to not duplicate this call.

Reply:
As I know this will result only one read from the storage, because of the overlay layer.
The logic was to not include the WitnessData check into do_* function, clients might not need it.
Other option could be to pass the actual rank as an argument, but since this function is public, I believe it must be sure (query by itself) about the actual rank, otherwise it might lead to the inconsistency.  
@bkchr what you think?

Reply:
i think it is probably fine